### PR TITLE
Expose handler; support skip or error on unknown behavior

### LINKED ISSUE #5: Expose RequestHandler and T
I've migrated Jaeger client to use this repo for client code, but had to leave a couple pieces untouched because the similar functionality is private in crossdock.
1. I cannot use `crossdock.Start()` directly because I need a way to configure some private members of the client class. Any objection to exposing the `requestHandler` as public?
2. Any objection to exposing a constructor for `entryT`? For example, I cannot unit test [this function](https://github.com/uber/jaeger-client-go/blob/master/crossdock/client/trace.go#L91) without it.

### PR DESCRIPTION
**Pull request by yurishkuro**
Fixes #5

@breerly @abhinav

### CODE REVIEW

**Comment by yurishkuro** [comment]
ready for review

File: client.go [hunk 1]
```diff
@@ -38,12 +38,20 @@ const BehaviorParam = "behavior"
...
+// If failOnUnknown is true, a behavior that is not in the behaviors map will cause the
+// handler to return an error status. Otherwise, the handler returns skipped status.
+func Handler(behaviors Behaviors, failOnUnknown bool) http.Handler {
+	return requestHandler{behaviors: behaviors, failOnUnknown: failOnUnknown}
+}
+
 type requestHandler struct {
-	behaviors Behaviors
+	behaviors     Behaviors
+	failOnUnknown bool
```
Comment (HelloGrayson) [inline client.go:54]: Almost would like to be more specific here and call this `failOnUnknownBehavior`... I looked at this for awhile before realizing what it was.

**Comment by HelloGrayson** [comment]
:shipit: 

(ty for the tests)

**Comment by HelloGrayson** [comment]
Merging to unblock @yurishkuro.

@abhinav if you have feedback we can follow up with issues.

File: client_test.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	defer server.Close()
+
+	verifyBehavior(t, server.URL, "b1", map[string]string{
+		"status": "passed",
+		"output": "ok",
+	})
+
+	verifyBehavior(t, server.URL, "b2", map[string]string{
+		"status": status,
+		"output": "unknown behavior \"b2\"",
```
Comment (abhinav) [inline client_test.go:65]: Nit: Use backticks to avoid escaping `"`

```
"output": `unknown behavior "b2"`
```

File: client_test.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+		{false, "skipped"},
+		{true, "failed"},
+	}
+
+	for _, test := range tests {
+		testHandler(t, test.failOnUnknown, test.status)
+	}
+}
+
+func testHandler(t *testing.T, failOnUnknown bool, status string) {
```
Comment (abhinav) [inline client_test.go:48]: Nit: Can you rename this to something like `runTestCase` or something? The test is already called `TestHandler`.

**Comment by abhinav** [comment]
LGTM with nits

**Pull request merged**